pkg/v3/plugin/hooks/add_from_staging.go

```diff
@@ -49,16 +50,40 @@ func (hook *AddFromStagingHook) RunHook(obs *ocr2keepersv3.AutomationObservation
 	}
 
 	results = hook.sorter.orderResults(results, rSrc)
-	if n := len(results); n > limit {
-		results = results[:limit]
-		hook.logger.Printf("skipped %d available results in staging", n-limit)
-	}
-	hook.logger.Printf("adding %d results to observation", len(results))
-	obs.Performable = append(obs.Performable, results...)
+	added, _ := hook.addByPercentageExceeded(obs, limit, results, ocr2keepersv3.ObservationPerformablesLimit)
+
+	hook.logger.Printf("skipped %d available results in staging", len(results)-added)
+
+	hook.logger.Printf("adding %d results to observation", added)
 
 	return nil
 }
 
+func (hook *AddFromStagingHook) addByPercentageExceeded(obs *ocr2keepersv3.AutomationObservation, limit int, results []automation.CheckResult, baseSize int) (int, int) {
+	if limit > len(results) {
+		limit = len(results)
+	}
+	existingPerformables := obs.Performable
+
+	obs.Performable = append(obs.Performable, results[:limit]...)
```

Comment:
i don't fully follow the logic of why we are appending here instead of just setting the observables. 

It would work directly if we were setting 
`obs.Performable = results[:limit]` ?

Reply:
The previous functionality appended performables, and we had tests that expected the append vs a set so I kept it this way. Happy to change this though!